qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/FaissSQIT.java

```diff
@@ -0,0 +1,273 @@
+            int k = 4;
+            Response searchResponse = searchKNNIndex(testIndex, new KNNQueryBuilder(TEST_FIELD, queryVector, k), k);
+            List<KNNResult> results = parseSearchResponse(EntityUtils.toString(searchResponse.getEntity()), TEST_FIELD);
+            assertEquals(k, results.size());
+            for (int i = 0; i < k; i++) {
+                assertEquals(k - i, Integer.parseInt(results.get(i).getDocId()));
+            }
+            deleteKNNIndex(testIndex);
+            validateGraphEviction();
+        }
+    }
+
+    public void testIVFSQFP16_onUpgradeWhenIndexedAndQueried_thenSucceed() throws Exception {
+        if (!isRunningAgainstOldCluster()) {
+            int dimension = 128;
+            int numDocs = 100;
+
+            // Add training data
+            createBasicKnnIndex(TRAIN_INDEX, TRAIN_TEST_FIELD, dimension);
+            int trainingDataCount = 200;
+            bulkIngestRandomVectors(TRAIN_INDEX, TRAIN_TEST_FIELD, trainingDataCount, dimension);
+
+            XContentBuilder builder = XContentFactory.jsonBuilder()
+                .startObject()
+                .field(NAME, METHOD_IVF)
+                .field(KNN_ENGINE, FAISS_NAME)
+                .field(METHOD_PARAMETER_SPACE_TYPE, "l2")
+                .startObject(PARAMETERS)
+                .startObject(METHOD_ENCODER_PARAMETER)
+                .field(NAME, ENCODER_SQ)
+                .startObject(PARAMETERS)
+                .field(FAISS_SQ_TYPE, FAISS_SQ_ENCODER_FP16)
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject();
+            Map<String, Object> method = xContentBuilderToMap(builder);
+
+            trainModel(TEST_MODEL, TRAIN_INDEX, TRAIN_TEST_FIELD, dimension, method, "faiss ivf sqfp16 test description");
+
+            // Make sure training succeeds after 30 seconds
+            assertTrainingSucceeds(TEST_MODEL, 30, 1000);
+
+            // Create knn index from model
+            String indexMapping = XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("properties")
+                .startObject(TEST_FIELD)
+                .field("type", "knn_vector")
+                .field(MODEL_ID, TEST_MODEL)
+                .endObject()
+                .endObject()
+                .endObject()
+                .toString();
+
+            createKnnIndex(testIndex, getKNNDefaultIndexSettings(), indexMapping);
+
+            indexTestData(testIndex, TEST_FIELD, dimension, numDocs);
+            queryTestData(testIndex, TEST_FIELD, dimension, numDocs);
+            deleteKNNIndex(TRAIN_INDEX);
+            deleteKNNIndex(testIndex);
+            deleteModel(TEST_MODEL);
+            validateGraphEviction();
+        }
+    }
+
+    private void validateGraphEviction() throws Exception {
+        // Search every 5 seconds 14 times to confirm graph gets evicted
+        int intervals = 14;
```

Comment:
why 14? and 5 seconds?

Reply:
TBH, I'm not sure. I replicated this from FaissIT. May be for some tests with less number of docs the graphs gets evicted quickly. For some tests with more docs it might take more time. So, checking it multiple times in short duration.